### PR TITLE
Improve Documentation

### PR DESCRIPTION
**Pull request by LukasHiveMQ**
https://hivemq.kanbanize.com/ctrl_board/22/cards/12393/details/

Refactor documentation
Move options to includes
Remove Synopsis

### CODE REVIEW

File: docs/_docs/publish.md [hunk 1]
```diff
@@ -4,176 +4,51 @@ redirect_from: /docs/publish.html
...
--->
+***
 
-*** 
+## Simple Examples
 
-## Synopsis
-
-``` 
-mqtt pub    -t <topics> [-t <topics>]...
```
Comment (gitseti) [inline docs/_docs/publish.md:32]: For some commands this overview was removed, for others not, and for some there is just the overview like `mqtt hivemq export [COMMAND]`

**Review by gitseti — APPROVED** [review]

File: docs/_docs/publish.md (hunk 1 above)
Comment (LukasHiveMQ) [reply docs/_docs/publish.md:32]: My fault, I should have converted this PR to a draft. I am still working on it. 
TL; DR: Every command should now be:
- Title
- Small introduction text
- First glimpse (Command overview without any options/flags)
- Short example
- Options explained
- ...


The current concept is to remove every synopsis section as it does not add value without explanations of the flags and options. Instead, every command overview shall contain an options section with a sorted overview of all available flags and options. As a side effect, this also greatly improves documentation maintenance as the option tables can be shared across multiple pages. The HiveMQ commands page is still a bit odd, I think it is best to split it up like the shell section so subcommands are separated.

File: docs/_docs/shell/connect.md [hunk 2]
```diff
@@ -198,14 +116,13 @@ mqtt>
 
 ***
 
-> Connect a client which is identified by myClient on specific settings and disconnect it afterwards
+Connect a client which is identified by myClient on specific settings and disconnect it afterward
```
Comment (tlopesPT) [inline docs/_docs/shell/connect.md:119]: I'd stick with afterwards, but both seem correct

File: docs/_docs/shell/connect.md [hunk 3]
```diff
@@ -198,14 +116,13 @@ mqtt>
...
 
 ```
 mqtt> con -i myClient -h broker.hivemq.com -V 3
 myClient@localhost> exit  # client is still connected
 mqtt> dis -i myClient -h broker.hivemq.com
 ```
 
-> **NOTE**: Besides the **identifier** also **hostname** has to be given to uniquely identify the client.
+**NOTE**: Besides the **identifier** also **hostname** has to be given to uniquely identify the client.
```
Comment (tlopesPT) [inline docs/_docs/shell/connect.md:127]:
```suggestion
**NOTE**: When specifying the **identifier** to uniquely identify the client, the **hostname** must also be provided.
If you don't specify both, the default settings for these attributes will be used, which may lead to unexpected behavior.
```

File: docs/_docs/shell/disconnect.md [hunk 4]
```diff
@@ -58,13 +56,13 @@ mqtt>
...
+Connect a client which is identified by myClient on specific settings and disconnect it afterward
 
 ```
 mqtt> con -i myClient -h broker.hivemq.com
 myClient@localhost> exit  # client is still connected
 mqtt> dis -i myClient -h broker.hivemq.com
 ```
 
-> **NOTE**: Besides the **identifier** also the **hostname** has to be given to uniquely identify the client.
+**NOTE**: Besides the **identifier** also the **hostname** has to be given to uniquely identify the client.
```
Comment (tlopesPT) [inline docs/_docs/shell/disconnect.md:67]:
```suggestion
**NOTE**: When specifying the **identifier** to uniquely identify the client, the **hostname** must also be provided.
If you don't specify both, the default settings for these attributes will be used, which may lead to unexpected behavior.
```

File: docs/_docs/shell/subscribe.md [hunk 5]
```diff
@@ -4,49 +4,41 @@ redirect_from: /docs/shell/subscribe.html
...
+
 ***
 
-##  Options
+## Options
+
+| Option | Long Version        | Explanation                                                                                                                                                                                             | Default |
+|--------|---------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|
+| `-oc`  | `--outputToConsole` | If this flag is set the output will be printed to the console.                                                                                                                                          | `false` |
+| `-s`   | `--stay`            | The subscribe emulates the same behavior as the subscribe command in non-shell mode. <br> **NOTE**: the subscriptions will be unsubscribed afterwards. <br> To cancel the command simply press *Enter*. | `false` |
```
Comment (tlopesPT) [inline docs/_docs/shell/subscribe.md:27]:
```suggestion
| `-s`   | `--stay`            | The subscribe emulates the same behavior as the subscribe command in non-shell mode. <br> **NOTE**: the subscriptions will be unsubscribed afterwards. <br> To cancel the command press *Enter*. | `false` |
```

File: docs/_docs/shell/switch.md [hunk 6]
```diff
@@ -1,45 +1,44 @@
...
-
+| Parameter Name | Explanation                                                                                                                                                                       | Examples                                                                                                      |
+|----------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------|
+| `contextName`  | The context name of a client consisting of the the client identifier concatenated by a @ with the hostname. The hostname may be omitted and will be filled with the default host. | `myClient@localhost`  `client2@broker.hivemq.com` or simply the `clientID` (default @localhost will be added) |
```
Comment (tlopesPT) [inline docs/_docs/shell/switch.md:22]:
```suggestion
| `contextName`  | The context name of a client consisting of the client identifier concatenated by a @ with the hostname. The hostname may be omitted and will be filled with the default host. | `myClient@localhost`  `client2@broker.hivemq.com` or simply the `clientID` (default @localhost will be added) |
```

File: docs/_docs/subscribe.md [hunk 7]
```diff
@@ -4,208 +4,119 @@ redirect_from: /docs/subscribe.html
...
+### Help Options
+
+{% include options/help-options.md defaultHelp=false %}
 
 *** 
 
 ## Further Examples
 
-> Subscribe to one topic with default QoS ``Exactly Once``
+Subscribe to one topic with default QoS `Exactly Once`.
```
Comment (tlopesPT) [inline docs/_docs/subscribe.md:76]:
```suggestion
Subscribe to one topic with QoS level `Exactly Once`.
```

File: docs/_docs/subscribe.md [hunk 8]
```diff
@@ -4,208 +4,119 @@ redirect_from: /docs/subscribe.html
...
+mqtt sub -t topic1 -q 0 -t topic2 -q 1 -t topic3 -q 2
 ```
 
 ***
 
-> Subscribe to a topic and output the received publish messages to the file ``publishes.log`` in the current directory
+Subscribe to a topic and output the received publish messages to the file `publishes.log` in the current directory.
 
-> **NOTE**: If the file is not created yet it will be created by the CLI. If it is present the received publish messages will be appended to the file.
+**NOTE**: If the file is not created yet, it will be created by the CLI. If it is present the received publish
```
Comment (tlopesPT) [inline docs/_docs/subscribe.md:97]:
```suggestion
**NOTE**: CLI creates the file if it does not exist. The received publish messages will be appended.
```

File: docs/_docs/subscribe.md [hunk 9]
```diff
@@ -4,208 +4,119 @@ redirect_from: /docs/subscribe.html
...
+mqtt sub -t topic -of publishes.log
 ```
 
 ***
 
-> Subscribe to a topic and output the received publish messages to the file ``publishes.log`` in a specified ``/usr/local/var`` directory
+Subscribe to a topic and output the received publish messages to the file `publishes.log` in a
+specified `/usr/local/var` directory.
+
+**NOTE**: If the file is not created yet, it will be created by the CLI. If it is present the received publish messages
```
Comment (tlopesPT) [inline docs/_docs/subscribe.md:109]:
```suggestion
**NOTE**: CLI creates the file if it does not exist. The received publish messages will be appended.
```

File: docs/_docs/subscribe/connect_options.md [hunk 10]
```diff
@@ -1,5 +1,5 @@
 ---
 title: Connect Options
-permalink: /docs/subscribe/#connect-options
+permalink: /docs/publish/#connect-options
```
Comment (tlopesPT) [inline docs/_docs/subscribe/connect_options.md:3]: is this link correct? there's a mix of /docs/publish and /docs/subscribe in the /subscribe directory

**Review by tlopesPT — CHANGES_REQUESTED** [review]

**Comment by tlopesPT** [comment]
still at 35/62 files reviewed but posting the comments to avoid github dropping them, beware of hidden conversations on this page
image

File: docs/_docs/swarm/run.md [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+| stop    | See [stop](#stop-run)   |
+
+***
+
+## Options
+
+| Option | Long Version | Explanation                                | Default                 |
+|--------|--------------|--------------------------------------------|-------------------------|
+| `-url` |              | The URL of the HiveMQ Swarm Rest endpoint. | `http://localhost:8888` |
+| `-r`   | `--run-id`   | The id of the run to stop.                 | `The current run`       |
```
Comment (tlopesPT) [inline docs/_docs/swarm/run.md:32]:
```suggestion
| `-r`   | `--run-id`   | The id of the run.                 | `The current run`       |
```
if I understand this correctly, you can set the run-id on start, and use it to stop a specific run as well, right?

File: docs/_docs/swarm/run.md [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+
+***
+
+## Options
+
+| Option | Long Version | Explanation                                           | Default                 |
+|--------|--------------|-------------------------------------------------------|-------------------------|
+| `-url` |              | The URL of the HiveMQ Swarm Rest endpoint.            | `http://localhost:8888` |
+| `-f`   | `--file`     | The scenario to execute.                              | required                |
+| `-r`   | `--run-id`   | The id of the run to stop.                            | `The current run`       |
```
Comment (tlopesPT) [inline docs/_docs/swarm/run.md:62]:
```suggestion
| `-r`   | `--run-id`   | The id of the run to start.                            | `The current run`       |
```
I don't see this option in the CLI, can you actually name a run on start?

File: docs/_docs/test.md [hunk 13]
```diff
@@ -3,153 +3,120 @@ nav_order: 6
...
 
 *** 
 
 ## Further Examples
 
-> Perform tests for MQTT 5 only
+Perform tests for MQTT 5 only
 
 ```
 $ mqtt test -h broker.hivemq.com -a -V 5
```
Comment (tlopesPT) [inline docs/_docs/test.md:92]: if `-a` means test all versions, doesn't adding -V 5 contradict this?

File: docs/_includes/options/authentication-options.md [hunk 14]
```diff
@@ -0,0 +1,6 @@
...
+|------------|-----------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------------------------------------------------|
+| `-u`       | `--user`        | Define the username for authentication.                                                                                                                                                                                                                                                              |                                                    |
+| `-pw`      | `--password`    | Define the password for authentication directly. <br> If left blank the user will be prompted for the password in console.                                                                                                                                                                           |                                                    |
```
Comment (tlopesPT) [inline docs/_includes/options/authentication-options.md:4]:
```suggestion
| `-pw`      | `--password`    | Define the password for authentication directly. <br> If left blank the user will be prompted for the password in the console.                                                                                                                                                                           |                                                    |
```

File: docs/_includes/options/authentication-options.md [hunk 15]
```diff
@@ -0,0 +1,6 @@
...
+| `-pw:env`  |                 | Define that the password for authentication is read in from an environment variable.                                                                                                                                                                                                                 | `MQTT_CLI_PW` if option is specified without value |
```
Comment (tlopesPT) [inline docs/_includes/options/authentication-options.md:5]:
```suggestion
| `-pw:env`  |                 | Define that the password for authentication is read from an environment variable.                                                                                                                                                                                                                 | `MQTT_CLI_PW` if the option is specified without a value |
```

File: docs/_includes/options/authentication-options.md [hunk 16]
```diff
@@ -0,0 +1,6 @@
...
+| `-pw:file` |                 | Define the path to a file from which the password is read in.                                                                                                                                                                                                                                        |                                                    |
```
Comment (tlopesPT) [inline docs/_includes/options/authentication-options.md:6]:
```suggestion
| `-pw:file` |                 | Define the path to a file from which the password is read from.                                                                                                                                                                                                                                        |                                                    |
```

File: docs/_docs/test.md [hunk 17]
```diff
@@ -3,153 +3,120 @@ nav_order: 6
...
 
-*** 
+## Options
 
-## Synopsis
+| Option | Long Version    | Explanation                                                                    | Default                      |
+|--------|-----------------|--------------------------------------------------------------------------------|------------------------------|
+| `-V`   | `--mqttVersion` | The MQTT version to test the broker on.                                        | Both versions will be tested |
+| `-a`   | `--all`         | Perform all tests for all MQTT versions                                        | `false` (Only test MQTT 3)   |
+| `-t`   | `--timeOut`     | The time to wait for the broker to respond (in seconds).                       | `10s`                        |
```
Comment (tlopesPT) [inline docs/_docs/test.md:57]: adding the `s` to 10s doesn't seem consistent (connect keep alive is 60, without s) or needed, but we may not want to change the behaviour at this point

File: docs/_includes/options/connect-options.md [hunk 18]
```diff
@@ -0,0 +1,13 @@
...
+| `-p`        | `--port`                  | The MQTT port.                                                                   | `1883`                             |
+| `-V`        | `--mqttVersion`           | The MQTT version can be set to 3 or 5.                                           | `5`                                |
+| `-i`        | `--identifier`            | A unique client identifier can be defined.                                       | A randomly generated UTF-8 String. |
+| `-ip`       | `--identifierPrefix`      | The prefix for randomly generated client identifiers, if no identifier is given. | `mqttClient`                       |
+| `-c`        | `--[no-]cleanStart`       | Whether the client should start a clean session.                                 | `true`                             |
+| `k`         | `--keepAlive`             | The keep alive of the client (in seconds)                                        | `60`                               |
+| `-se`       | `--sessionExpiryInterval` | Session expiry value in seconds.                                                 | `0` (Instant Expiry)               |
```
Comment (tlopesPT) [inline docs/_includes/options/connect-options.md:10]:
```suggestion
| `-se`       | `--sessionExpiryInterval` | The session expiry value (in seconds).                                                 | `0` (Instant Expiry)               |
```

File: docs/_includes/options/connect-options.md [hunk 19]
```diff
@@ -0,0 +1,13 @@
...
+| `-h`        | `--host`                  | The MQTT host.                                                                   | `localhost`                        |
+| `-p`        | `--port`                  | The MQTT port.                                                                   | `1883`                             |
+| `-V`        | `--mqttVersion`           | The MQTT version can be set to 3 or 5.                                           | `5`                                |
+| `-i`        | `--identifier`            | A unique client identifier can be defined.                                       | A randomly generated UTF-8 String. |
+| `-ip`       | `--identifierPrefix`      | The prefix for randomly generated client identifiers, if no identifier is given. | `mqttClient`                       |
+| `-c`        | `--[no-]cleanStart`       | Whether the client should start a clean session.                                 | `true`                             |
+| `k`         | `--keepAlive`             | The keep alive of the client (in seconds)                                        | `60`                               |
```
Comment (tlopesPT) [inline docs/_includes/options/connect-options.md:9]:
```suggestion
| `k`         | `--keepAlive`             | The keep alive of the client (in seconds).                                        | `60`                               |
```

File: docs/_includes/options/connect-options.md [hunk 20]
```diff
@@ -0,0 +1,13 @@
...
+| `-se`       | `--sessionExpiryInterval` | Session expiry value in seconds.                                                 | `0` (Instant Expiry)               |
+| `-Cup`      | `--connectUserProperty`   | A user property of the connect message.                                          |                                    |
+| `--ws`      |                           | Use WebSocket transport protocol.                                                | `false`                            |
+| `--ws:path` |                           | The path to the WebSocket located at given broker host.                          |                                    |
```
Comment (tlopesPT) [inline docs/_includes/options/connect-options.md:13]:
```suggestion
| `--ws:path` |                           | The path to the WebSocket located at the given broker host.                          |                                    |
```

File: docs/_includes/options/connect-restrictions-options.md [hunk 21]
```diff
@@ -0,0 +1,10 @@
+| Option | Long Version            | Explanation                                                                                                       | Default     |
+|--------|-------------------------|-------------------------------------------------------------------------------------------------------------------|-------------|
+|        | `--rcvMax`              | The maximum amount of not acknowledged publishes with QoS 1 or 2 the client accepts from the server concurrently. | `65535`     |
+|        | `--sendMax`             | The maximum amount of not acknowledged publishes with QoS 1 or 2 the client send to the server concurrently.      | `65535`     |
```
Comment (tlopesPT) [inline docs/_includes/options/connect-restrictions-options.md:4]:
```suggestion
|        | `--sendMax`             | The maximum amount of not acknowledged publishes with QoS 1 or 2 the client sends to the server concurrently.      | `65535`     |
```

File: docs/_includes/options/debug-options.md [hunk 22]
```diff
@@ -0,0 +1,4 @@
+| Option | Long Version | Explanation                                         | Default |
+|--------|--------------|-----------------------------------------------------|---------|
+| `-d`   | `--debug`    | Print info level debug messages to the console.     | `false` |
```
Comment (tlopesPT) [inline docs/_includes/options/debug-options.md:3]:
```suggestion
| `-d`   | `--debug`    | Print debug messages to the console.     | `false` |
```

File: docs/_docs/shell/connect.md (hunk 2 above)
Comment (LukasHiveMQ) [reply docs/_docs/shell/connect.md:119]: Yeah also way more natural to me, but since all of our documentation is written in AE and IntelliJ also suggests using it, I changed it.

File: docs/_docs/subscribe/connect_options.md (hunk 10 above)
Comment (LukasHiveMQ) [reply docs/_docs/subscribe/connect_options.md:3]: Great catch. Definitely a bug.

File: docs/_includes/options/debug-options.md [hunk 23]
```diff
@@ -0,0 +1,4 @@
...
+| `-v`   | `--verbose`  | Print detailed debug level messages to the console. | `false` |
```
Comment (tlopesPT) [inline docs/_includes/options/debug-options.md:4]:
```suggestion
| `-v`   | `--verbose`  | Print trace messages to the console. | `false` |
```
not really sure if we want to change this description, up to you, but "info" and "debug" in the -d explanation is confusing since these are 2 different log levels

File: docs/_docs/swarm/run.md (hunk 11 above)
Comment (LukasHiveMQ) [reply docs/_docs/swarm/run.md:32]: Yes!

File: docs/_includes/options/publish-options.md [hunk 24]
```diff
@@ -0,0 +1,14 @@
...
+|------------|----------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|
+| `-t`       | `--topic`                  | The MQTT topic to which the message will be published.                                                                                                                                                                                |         |
+| `-m`       | `--message`                | The message which will be published on the topic.                                                                                                                                                                                     |         |
+| `-m:file`  | `--message-file`           | The file whose payload will be published on the topic.                                                                                                                                                                                |         |
```
Comment (tlopesPT) [inline docs/_includes/options/publish-options.md:5]:
```suggestion
| `-m:file`  | `--message-file`           | The file containing the payload which will be published on the topic.                                                                                                                                                                                |         |
```

File: docs/_includes/options/publish-options.md [hunk 25]
```diff
@@ -0,0 +1,14 @@
...
+| `-m:empty` | `--message-empty`          | Sets the message to an empty payload.                                                                                                                                                                                                 |         |
+| `-r`       | `--[no-]retain`            | Whether the message will be retained.                                                                                                                                                                                                 | `False` |
```
Comment (tlopesPT) [inline docs/_includes/options/publish-options.md:7]:
```suggestion
| `-r`       | `--[no-]retain`            | Whether the message will be retained.                                                                                                                                                                                                 | `false` |
```

File: docs/_docs/swarm/run.md (hunk 11 above)
Comment (LukasHiveMQ) [reply docs/_docs/swarm/run.md:32]: You are correct. Will remove the row

File: docs/_docs/swarm/run.md (hunk 12 above)
Comment (LukasHiveMQ) [reply docs/_docs/swarm/run.md:62]: Will also remove that row.

File: docs/_includes/options/subscribe-options.md [hunk 26]
```diff
@@ -0,0 +1,9 @@
...
+|--------|------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|
+| `-t`   | `--topic`        | The MQTT topic the client will subscribe to.                                                                                                                                                                                          |         |
+| `-q`   | `--qos`          | Define the quality of service level. If only one QoS is specified it will be used for all topics.<br> You can define a specific QoS level for every topic. The corresponding QoS levels will be matched in order to the given topics. | `0`     |
+| `-of`  | `--outputToFile` | If a file is given print the received publishes to the specified output file. If the file is not present it will be created.                                                                                                          |         |
```
Comment (tlopesPT) [inline docs/_includes/options/subscribe-options.md:5]:
```suggestion
| `-of`  | `--outputToFile` | Append the received publish messages to a file. Creates the file if it does not exist.                                                                                                           |         |
```

File: docs/_includes/options/will-options.md [hunk 27]
```diff
@@ -0,0 +1,13 @@
...
+| `-Wq`  | `--willQualityOfService`       | QoS level of the will message.                                                                 | `0`                        |
+| `-Wr`  | `--willRetain`                 | Retain the will message.                                                                       | `false`                    |
+| `-Wt`  | `--willTopic`                  | Topic of the will message.                                                                     |                            |
+| `-Wcd` | `--willCorrelationData`        | Correlation data of the will message                                                           |                            |
+| `-Wct` | `--willContentType`            | Description of the will message's content.                                                     |                            |
+| `-Wpf` | `--willPayloadFormatIndicator` | Payload format can be explicitly specified as `UTF8` else it may be `UNSPECIFIED`.             |                            |
+| `-Wrt` | `--willResponseTopic`          | Topic Name for a response message.                                                             |                            |
```
Comment (tlopesPT) [inline docs/_includes/options/will-options.md:12]:
```suggestion
| `-Wrt` | `--willResponseTopic`          | Topic name for a response message.                                                             |                            |
```

File: docs/_docs/test.md (hunk 13 above)
Comment (LukasHiveMQ) [reply docs/_docs/test.md:92]: The test is for all tests in the MQTT 5 scope. -a enables additional tests for MQTT 5. The description of -a is somewhat misleading.

**Review by tlopesPT — APPROVED** [review]

File: docs/_docs/test.md (hunk 17 above)
Comment (LukasHiveMQ) [reply docs/_docs/test.md:57]: In code it its without the s. We can change it inside the documentation.
